src/Views/HardwareView.vala

```diff
@@ -105,35 +120,51 @@ public class About.HardwareView : Gtk.Grid {
             attach (manufacturer_info, 0, 2, 2, 1);
 
             if (product_name != null) {
-                product_name_info.label = product_name;
-            }
 
-            if (product_version != null) {
-                var product_version_info = new Gtk.Label ("(" + product_version + ")");
-                product_version_info.get_style_context ().add_class (Gtk.STYLE_CLASS_DIM_LABEL);
-                product_version_info.set_selectable (true);
-                product_version_info.ellipsize = Pango.EllipsizeMode.END;
-                product_version_info.xalign = 0;
-                product_name_info.xalign = 1;
-                attach (product_name_info, 0, 1, 1, 1);
-                attach (product_version_info, 1, 1, 1, 1);
-            } else {
-                attach (product_name_info, 0, 1, 2, 1);
+                if (product_version != null) {
+                    var product_version_info = new Gtk.Label ("(" + product_version + ")");
+                    product_version_info.get_style_context ().add_class (Gtk.STYLE_CLASS_DIM_LABEL);
+                    product_version_info.set_selectable (true);
+                    product_version_info.ellipsize = Pango.EllipsizeMode.END;
+                    product_version_info.xalign = 0;
+                    product_name_info.xalign = 1;
+                    attach (product_name_info, 0, 1, 1, 1);
+                    attach (product_version_info, 1, 1, 1, 1);
+                } else {
+                    attach (product_name_info, 0, 1, 2, 1);
+                }
             }
 
             if (manufacturer_support_url != null) {
                 var manufacturer_website_info = new Gtk.LinkButton.with_label (manufacturer_support_url, _("Manufacturer Website"));
-                attach (manufacturer_website_info, 0, 7, 2, 1);
+                attach (manufacturer_website_info, 0, 9, 2, 1);
             }
         } else {
             attach (product_name_info, 0, 1, 2, 1);
         }
     }
 
+    private string clean_hostname (string hostname) {
+        string cleaned_hostname = hostname.replace (" ", "-");
+        return cleaned_hostname;
+    }
+
+    private void change_hostname (string hostname) {
+        if (hostname.length == 0) {
+            hostname = "localhost";
+        }
+
+        try {
+            Process.spawn_command_line_async ("hostnamectl set-hostname %s".printf(hostname.to_ascii ()));
```

Comment:
That's not the right way to do it. The right way is to use the `org.freedesktop.hostname1` D-Bus interface on the system bus. This also allows to make sure that the user actually has the right to change the hostname (which should be shown in the interface too)